servers/services/src/main/java/org/projectnessie/services/rest/TreeResource.java

```diff
@@ -287,14 +299,25 @@ public void mergeRefIntoBranch(String branchName, String hash, Merge merge)
           .merge(
               toHash(merge.getFromRefName(), merge.getFromHash(), true).get(),
               BranchName.of(branchName),
-              toHash(hash, true));
+              toHash(hash, true),
+              updateMetaWithMergePropsFunc(MergeOp.MERGE, merge.getFromRefName()));
     } catch (ReferenceNotFoundException e) {
       throw new NessieNotFoundException(e.getMessage(), e);
     } catch (ReferenceConflictException e) {
       throw new NessieConflictException(e.getMessage(), e);
     }
   }
 
+  private BiFunction<Serializer<CommitMeta>, ByteString, ByteString> updateMetaWithMergePropsFunc(
+      MergeOp mergeOp, String fromRefName) {
+    return (ser, in) -> {
+      CommitMeta commitMeta = ser.fromBytes(in);
+      String mergeProp = String.format("%s_ref", mergeOp.name().toLowerCase());
```

Comment:
Let's not add that property using this PR. It's a different scope and we are not yet sure whether it should be added at all.

Reply:
ok